### PR TITLE
Feat/pci kubernetes view cluster refresh tapc 1516 tapc 1752

### PR DESCRIPTION
**Pull request by ppprevost**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? |no
| Tickets          | [TACP-1752, TAPC-1516]
| License          | BSD 3-Clause

- [ ] Try to keep pull requests small so they can be easily reviewed.
- [ ] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [ ] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts [hunk 1]
```diff
@@ -39,10 +39,22 @@ export const getAllKubeQueryKey = (projectId: string) => [
...
+        return false;
+      }
+      const notAllReady = query.state.data?.some(
+        (data) => data.status !== 'READY',
+      );
+      if (notAllReady) {
+        return refetchIntervalTime;
+      }
+      return false;
+    },
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts:57]: As the `refetchInterval` function is declared twice (here and in `useKubernetesCluster`), you can create a dedicated function that takes the predicate function as parameter 😌.

And, additionally, if you want to be compliant with inversion of dependencies principle, you can give the function as the hook parameter 😎

File: packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.spec.ts [hunk 2]
```diff
@@ -63,6 +63,53 @@ describe('useKubernetesCluster', () => {
...
+    );
+
+    const { result } = renderHook(
+      () => useKubernetesCluster(projectId, kubeId, refetchIntervalTime),
+      { wrapper },
+    );
+
+    await waitFor(() => expect(result.current.isSuccess).toBe(true));
+    vi.useFakeTimers({ shouldAdvanceTime: true });
+    await waitFor(() =>
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.spec.ts:109]: I am not sure that the async `waitFor` block is mandatory here as you have already waited for the first assertion in line 107.

**Comment by Eric-ciccotti** [comment]
TU ko

**Comment by fredericvilcot** [comment]
I have a global interrogation about the behavior, what's about if the cluster in on error state? 
Will there be an infinite refetch, or at least until the cluster has been rescued? 
It's an open question about the global polling mechanism for statuses :)

**Pull request closed without merging**